x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -827,6 +843,33 @@ private Join resolveLookupJoin(LookupJoin join) {
             }
         }
 
+        private boolean isCompletelyRightSideAndTranslationAware(Expression expression, AttributeSet rightOutputSet) {
+            // Check if all references in the expression are from the right side
+            boolean isCompletelyRightSide = rightOutputSet.containsAll(expression.references());
+
+            if (isCompletelyRightSide == false) {
+                return false;
+            }
+
+            // Check if the expression and all its subexpressions implement TranslationAware
+            // and are translatable to Lucene
+            return isTranslationAware(expression);
+        }
+
+        private boolean isTranslationAware(Expression expression) {
+            // Check if the expression itself implements TranslationAware
+            if (expression instanceof TranslationAware == false) {
+                return false;
+            }
+
+            // Check if the expression is translatable
+            TranslationAware.Translatable translatable = translatable(expression, LucenePushdownPredicates.DEFAULT);
+            if (translatable == TranslationAware.Translatable.NO) {
+                return false;
+            }
+            return true;
```

Comment:
This is a little verbose: comments that just state what the next line of code does, and early returns that are drawn over multiple lines and can be compacted (esp. isCompletelyRightSideAndTranslationAware, consider a ternary there). Looks like maybe AI helped write this? Maybe let's clean this up a little :)